File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        string temp=s;
//         stack<char>st;
//         // int len=0;
        
//         for(auto ch:s)
//         { 
            
//             if(ch==' ')
//             {
//                 if(!st.empty())
//                 {
//                     // len=0;
//                 while(!st.empty())
//                 {
//                     st.pop();
//                 }
//                 }
//             }
       
        
//             else
//             {
//                 st.push(ch);
//                 // len++;
//             }
           
//     }
        
       
      
        
//           if(st.size()==0)
//           {
              
//           }
//         else
//         {
//             return st.size();
//         }
        
        
        
        int count=0;
        
        for(int i=s.length()-1;i>=0;i--)
        {
            if(s[i]!=' ')
            {
                count++;

            }
            else
            {
                if(count>0)
                {
                    return count;
                }
            }
            
        }
       
        return count;
        
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        
        int i=s.length()-1;
        while(s.charAt(i)==' ')
        {
            i--;
        }
        int count=0;
        while(s.charAt(i)!=' ')
        {
            count++;
            i--;
            if(i==-1)
            {
                break;
            }
        }
        return count;
        
    }
}